src/main/java/io/jenkins/plugins/signpath/SecretRetrieval/CredentialBasedSecretRetriever.java

```diff
@@ -40,12 +48,17 @@ public Secret retrieveSecret(String id) throws SecretNotFoundException {
             throw new SecretNotFoundException(String.format("The secret '%s' could not be found in the credential store.", id));
         }
 
-        if (credential.getScope() != CredentialsScope.SYSTEM) {
-            CredentialsScope scope = credential.getScope();
-            String scopeName = scope == null ? "<null>" : scope.getDisplayName();
+        CredentialsScope credentialScope = credential.getScope();
+        
+        if (allowedScopes.length > 0 && !Arrays.asList(allowedScopes).contains(credentialScope)) {
```

Comment:
Is the `allowedScope.length > 0` actually necessary?

Reply:
If allowedScope is empty there is no scope restriction